pkg/log/logshipper/logshipper.go

```diff
@@ -0,0 +1,127 @@
+package logshipper
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/kolide/launcher/pkg/sendbuffer"
+)
+
+// TODO: consolidate this var, but where?
+var observabilityIngestTokenKey = []byte("observability_ingest_auth_token")
+
+const (
+	truncatedFormatString = "%s[TRUNCATED]"
+)
+
+type LogShipper struct {
+	sender            *authedHttpSender
+	sendBuffer        *sendbuffer.SendBuffer
+	logger            log.Logger
+	knapsack          types.Knapsack
+	stopFunc          context.CancelFunc
+	isShippingEnabled bool
+}
+
+func New(k types.Knapsack) (*LogShipper, error) {
+	token, _ := k.TokenStore().Get(observabilityIngestTokenKey)
+
+	logEndpoint, err := logEndpoint(k)
+	if err != nil {
+		return nil, err
+	}
+
+	sender := newAuthHttpSender(logEndpoint, string(token))
+
+	sendInterval := time.Minute * 1
+	if k.Debug() {
+		sendInterval = time.Second * 1
+	}
+
+	sendBuffer := sendbuffer.New(sender, sendbuffer.WithSendInterval(sendInterval))
+	logger := log.NewJSONLogger(sendBuffer)
+
+	return &LogShipper{
+		sender:     sender,
+		sendBuffer: sendBuffer,
+		logger:     logger,
+		knapsack:   k,
+	}, nil
+}
+
+func (ls *LogShipper) Ping() {
+	token, _ := ls.knapsack.TokenStore().Get(observabilityIngestTokenKey)
+	ls.sender.authtoken = string(token)
+
+	endpoint, err := logEndpoint(ls.knapsack)
+	if err != nil {
+		ls.logger.Log("msg", "failed to get endpoint", "err", err)
+		return
+	}
+	ls.sender.endpoint = endpoint
+	ls.isShippingEnabled = ls.knapsack.LogIngestServerURL() != ""
+
+	if !ls.isShippingEnabled {
+		ls.sendBuffer.DeleteAllData()
+	}
+
+}
+
+func (ls *LogShipper) Run() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	ls.stopFunc = cancel
+	return ls.sendBuffer.Run(ctx)
+}
+
+func (ls *LogShipper) Stop(_ error) {
+	ls.stopFunc()
+}
+
+func logEndpoint(k types.Knapsack) (string, error) {
+	endpoint := k.LogIngestServerURL()
+
+	if !strings.HasPrefix(endpoint, "http") {
+		scheme := "https"
+		if k.DisableObservabilityIngestTLS() {
```

Comment:
Hm I'm not sure we should be reusing this flag? If we don't think the log vs trace ingest URL will be the same in the future, we are going to want to be able to disable TLS for them separately.

Reply:
I am unsure we should have this flag at all. It seems better baked into the scheme on the endpoint. Though maybe we need one for whether or not to validate the certificates.

Reply:
dropped usage of flag and just rely on endpoint given by server